core/domain/user_services.py

```diff
@@ -67,14 +67,17 @@ class UserSettings(object):
         preferred_language_codes: list(str) or None. Exploration language
             preferences specified by the user.
         preferred_site_language_code: str or None. System language preference.
+        role: str. Role of the user. Role corresponds to actions user can
+            perform.
     """
     def __init__(
             self, user_id, email, username=None, last_agreed_to_terms=None,
             last_started_state_editor_tutorial=None, last_logged_in=None,
             last_created_an_exploration=None,
             last_edited_an_exploration=None, profile_picture_data_url=None,
             user_bio='', subject_interests=None, first_contribution_msec=None,
-            preferred_language_codes=None, preferred_site_language_code=None):
+            preferred_language_codes=None, preferred_site_language_code=None,
+            role='EXPLORATION_EDITOR'):
```

Comment:
That seems a bit odd. Can we make it mandatory, and specify it everywhere it's needed? I think `role` is different from the rest since it is pretty vital information (it determines all the permissions). Most of the other stuff is 'optional'.